state/postgresql/v1/migrations.go

```diff
@@ -39,15 +39,20 @@ func performMigrations(ctx context.Context, db pginterfaces.PGXPoolConn, opts po
 			opts.Logger.Infof("Creating state table '%s'", opts.StateTableName)
 			_, err := db.Exec(
 				ctx,
-				fmt.Sprintf(
-					`CREATE TABLE IF NOT EXISTS %s (
-							key text NOT NULL PRIMARY KEY,
-							value jsonb NOT NULL,
-							isbinary boolean NOT NULL,
-							insertdate TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
-							updatedate TIMESTAMP WITH TIME ZONE NULL
-						)`,
-					opts.StateTableName,
+				fmt.Sprintf(`
+DO $$
+BEGIN
+	IF NOT EXISTS (SELECT 1 FROM information_schema.tables WHERE table_name = '%[1]s') THEN
```

Comment:
This won't work if the table name contains a schema. You need to check if there's a schema (`schema.table`) and check for the `table_schema` column too

Reply:
I've adjusted a bit to capture this error better. So we don't need to acct for things like this anymore :)